### PR TITLE
Get rid of MemoryDB denote

### PR DESCRIPTION
**Pull request by arkpar**
Reduces memory footprint by at least 40% when syncing

### CODE REVIEW

File: util/src/trie/journal.rs [hunk 1]
```diff
@@ -52,16 +52,16 @@ impl Journal {
...
 			let rlp_sha3 = rlp.sha3();
 
-			trace!("new_node: reference node {:?} => {:?}", rlp_sha3, rlp.pretty());
+			trace!("new_node: reference node {:?} => {:?}", rlp_sha3, &*rlp);
 			out.append(&rlp_sha3);
 			self.0.push(Operation::New(rlp_sha3, rlp));
 		}
 		else {
-			trace!("new_node: inline node {:?}", rlp.pretty());
+			trace!("new_node: inline node {:?}", &*rlp);
```
Comment (rphmeier) [inline util/src/trie/journal.rs]: lost the pretty-printing here

File: util/src/trie/mod.rs [hunk 2]
```diff
@@ -76,7 +76,7 @@ impl fmt::Display for TrieError {
 pub type Result<T> = ::std::result::Result<T, Box<TrieError>>;
 
 /// Trie-Item type.
-pub type TrieItem<'a> = Result<(Vec<u8>, &'a [u8])>;
+pub type TrieItem<'a> = Result<(Vec<u8>, DBValue)>;
```
Comment (rphmeier) [inline util/src/trie/mod.rs]: lifetime here isn't necessary anymore.

**Comment by gavofyork** [comment]
gitlab fails?

**Comment by arkpar** [comment]
gitlab `js-test` fails. Does not seem relevant

**Pull request merged**